server/coordinator/procedure/manager_impl.go

```diff
@@ -175,15 +175,31 @@ func (m *ManagerImpl) startProcedureWorker(ctx context.Context, newProcedure Pro
 }
 
 // Whether a waiting procedure could be running procedure.
-func (m *ManagerImpl) checkValid(_ context.Context, _ Procedure, _ *metadata.ClusterMetadata) bool {
+func (m *ManagerImpl) checkValid(p Procedure, clusterMetadata *metadata.ClusterMetadata) bool {
```

Comment:
The `m` seems unused, so maybe we can let it as a free function for unit test friendly.